### PR TITLE
Change the default poetry extras

### PR DESCRIPTION
**Pull request by izellevy**
## Problem

By default we install all-extras in dockerfile that causes out of disk error while building the image. We need to shrink the image size.

## Solution

Now we install only the necessary extras that dropped the image size from 6GB to 600MB.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Infrastructure change (CI configs, etc)
- [ ] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

## Test Plan

Describe specific steps for validating this change.

### CODE REVIEW

File: .github/actions/install-deps-and-canopy/action.yml [hunk 1]
```diff
@@ -41,12 +36,8 @@ runs:
...
-      CANOPY_EXTRAS: ${{ inputs.extras }}
-    run: poetry install --no-interaction --no-root --with dev --extras "${CANOPY_EXTRAS}"
+    run: make install POETRY_INSTALL_EXTRA_ARGS="--no-interaction --no-root --with dev"
   - name: Install project
     if: ${{ inputs.install-canopy == 'true' }}
     shell: bash
-    env:
-      CANOPY_EXTRAS: ${{ inputs.extras }}
-    run: make install-extras
+    run: make install POETRY_INSTALL_EXTRA_ARGS="--with dev"
```
Comment (igiloh-pinecone) [inline .github/actions/install-deps-and-canopy/action.yml:43]: We should `--no-interaction` here as well

File: Makefile [hunk 2]
```diff
@@ -1,20 +1,24 @@
 TEST_WORKER_COUNT = 8
 
-REPOSITORY = ghcr.io/pinecone-io/canopy
+POETRY_INSTALL_SYSTEM_ARGS = -E cohere -E transformers
+POETRY_INSTALL_EXTRA_ARGS =
```
Comment (igiloh-pinecone) [inline Makefile:4]: The names here are really confusing, with `SYSTEM_ARGS` being a list of `extra`s, and `EXTRA_ARGS` being used for passing system arguments like `--no-root`...

Let's use something like `DEFAULT_EXTRAS` and `POETRY_INSTALL_ARGS`.

File: Makefile [hunk 3]
```diff
@@ -23,10 +27,7 @@ static:
 	poetry run mypy src
 
 install:
-	poetry install
```
Comment (igiloh-pinecone) [inline Makefile:26]: That's not the right approach. The default behavior for most users would still be to install canopy itself only.  

The `install-extras` is meant for our CI workflows, which we want to install all the "default" extras on top of base canopy (but not --all-extras`).

File: Makefile [hunk 4]
```diff
@@ -42,34 +43,41 @@ test-e2e:
...
 
 docker-run-dev:
-	docker run $(CONTAINER_COMMON_RUN_ARGS) $(CONTAINER_EXTRA_RUN_ARGS) -it $(REPOSITORY)-dev:$(IMAGE_TAG)
+	docker run $(CONTAINER_SYSTEM_RUN_ARGS) $(CONTAINER_EXTRA_RUN_ARGS) -it $(REPOSITORY)-dev:$(IMAGE_TAG)
+
+print-var:
+	@echo $($(VAR))
 
 help:
 	@echo "Available targets:"
```
Comment (igiloh-pinecone) [inline Makefile:64]: We should consider some pattern for automatically generating this help message by adding "docstrings" to the commands themselves.  
There are a few clever `Makefile` tricks that allow that (with some `awk` or `sed` or something).  

Not for this PR, just a general thought for the future

**Review by igiloh-pinecone — CHANGES_REQUESTED** [review]
@izellevy please see a few comments

File: .github/actions/install-deps-and-canopy/action.yml (hunk 1 above)
Comment (izellevy) [reply .github/actions/install-deps-and-canopy/action.yml]: Done

File: Makefile (hunk 3 above)
Comment (izellevy) [reply Makefile:26]: Changed it

**Review by acatav — APPROVED** [review]

**Pull request merged**